teensy/teensy_main/teensy_main.ino

```diff
@@ -391,7 +354,7 @@ void initial_hand_shake_sequence()
   }
 }
 
-void pull_data(){
+void sample_data_and_build_json_packet(){
```

Comment:
If you find yourself using the word `and` in a function name, it would probably be wise split the functionality into multiple functions.